src/components/hooks/useStarterPackEntry.native.ts

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+
+import {createStarterPackLinkFromAndroidReferrer} from 'lib/strings/starter-pack'
+import {isAndroid} from 'platform/detection'
+import {useHasCheckedForStarterPack} from 'state/preferences/used-starter-packs'
+import {useSetActiveStarterPack} from 'state/shell/starter-pack'
+import SwissArmyKnife from '../../../modules/expo-bluesky-swiss-army'
+import GooglePlayReferrer from '../../../modules/expo-google-play-referrer'
+
+export function useStarterPackEntry() {
```

Comment:
Very nit: naming here kinda odd to me, like it's called "entry" and it sets "active pack" and returns "has checked referrer". It's like "use init starter pack from referrer" and returns "is done" or "ready". Very nit, just spitballing.